monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js

```diff
@@ -276,12 +276,21 @@ class ConfigurePageComponent extends AuthComponent {
 
   setConfigOnImport = (event) => {
     try {
+      let parsedConfig = JSON.parse(event.target.result);
+
+      let importedAttackConfig = parsedConfig.attack;
+      delete parsedConfig.attack;
+      let importedMonkeyConfig = parsedConfig;
+
       this.setState({
-        configuration: JSON.parse(event.target.result),
+        configuration: importedMonkeyConfig,
+        attackConfig: importedAttackConfig,
         lastAction: 'import_success'
       }, () => {
         this.sendConfig();
-        this.setInitialConfig(JSON.parse(event.target.result))
+        this.matrixSubmit();
```

Comment:
The problem with this feature is that ATT&CK doesn't represent the actual configuration (there is no ATT&CK configuration), it represents changes made to the configuration using ATT&CK matrix. So if user changed ATT&CK matrix, then went to configuration and changed post breach actions it potentially created a conflict with ATT&CK matrix.
Since we got that out of the way (maybe we should refactor the ATT&CK matrix page to be more clear about this), it's clear that you should not apply ATT&CK matrix to the imported config (matrix is already applied to config and further changed). 
I got nothing against setting the state of ATT&CK matrix on import, but it shouldn't be applied to config.

Reply:
That's a good point!
Along with refactoring the ATT&CK matrix page to be more clear about this (open an issue?), we should change how this is reported, similar to how we changed it in [this PR](https://github.com/guardicore/monkey/pull/717).
Right now, if I enable the *Trap* technique in the ATT&CK config but disable it in the PBA config, the report gives me the message, "Monkey didn't use the trap command since it didn't run on any Linux machines," which is wrong, even though I ran it on a Linux machine. We ought to cross check with the Monkey config and say something like, "Monkey didn't use the trap command since it was disabled in the PBA config, which can be found under the *Monkey* tab in the configuration". I'll open an issue for this.

Anyhow, given the level of inconsistency that this would introduce in the configuration, I'd rather put it off until we fix the above issues. WDYT?

Reply:
> We ought to cross check with the Monkey config and say something like, "Monkey didn't use the trap command since it was disabled in the PBA config
If monkey decides upon the state of the ATT&CK technique only from ATT&CK matrix and not from config, that should be changed.

Reply:
> Anyhow, given the level of confusion and inconsistency that this would introduce in the configuration, I'd rather put it off until we fix the above issues. WDYT?

I think it's still a good change and if anything it brings more consistency, because you're getting the same config page state you exported - ATT&CK matrix included. Just don't "apply the matrix", only show it.

Reply:
Well then we should also have a way to tell the user that their clicking on "Submit" will change some other configuration as well.
Alternatively, we could make sure there's a two-way dependency for the techniques and PBAs. Right now, only the PBA is mapped to the respective attack technique (enabling/disabling the attack technique enables/disables the PBA, but the other way around does not happen).

Reply:
Good point, clicking submit after importing (if you're on ATT&CK screen) might change the config. Yes, we probably need a prompt for that case (where ATT&CK is applied to already modified config). Two way dependencies is a bit too complex to be worth it IMO.